subsys/bluetooth/mesh/main.c

```diff
@@ -462,6 +462,14 @@ int bt_mesh_suspend(void)
 
 	bt_mesh_access_suspend();
 
+	if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT)) {
+		(void)bt_mesh_pb_gatt_srv_disable();
```

Comment:
When resuming mesh, `bt_mesh_proxy_gatt_enable()` returns `1`. This is due to the return value of [k_work_schedule()](https://docs.zephyrproject.org/latest/kernel/services/threads/workqueue.html#c.k_work_schedule)  being returned.
This is not really an error and should be ignored?

Reply:
Suspending\resuming functionality shouldn't know implementation details of other modules. There are API functions. These should be sufficient. If API function behaves incorrectly, then it should be fixed inside the module. There shouldn't be workarounds here for adjusting internal implementation.
Also, suspending\resuming functionality shouldn't predict reasons of returned values\errors. Today it returns errors because device is unprovisioned, tomorrow somebody adds new functionality and new statuses.

Reply:
It now returns error codes. Added a separate commit after offline discussion fixing the return value of bt_mesh_proxy_gatt_enable `bt_mesh_proxy_gatt_enable`